k8s/plugin/src/resources/upgrade.rs

```diff
@@ -54,57 +61,101 @@ impl UpgradeResources {
         })
     }
 
-    /// Install the upgrade resources
-    pub async fn install(ns: &str) {
-        match UpgradeResources::new(ns).await {
-            Ok(uo) => {
-                let pp = PostParams::default();
-                // Create a service account
-                match uo.get_service_account().await {
-                    Ok(sa) => {
-                        if sa.iter().count() == 0 {
+    /// Create/Delete service action
+    pub async fn service_account_actions(&self, ns: &str, action: Actions) -> Result<(), ()> {
+        match self.get_service_account().await {
+            Ok(sa) => {
+                if sa.iter().count() == 0 {
+                    match action {
+                        Actions::Create => {
                             let ns = Some(ns.to_string());
                             let service_account = objects::upgrade_operator_service_account(
                                 ns,
-                                uo.release_name.clone(),
+                                self.release_name.clone(),
                             );
                             let pp = PostParams::default();
-                            match uo.service_account.create(&pp, &service_account).await {
+                            match self.service_account.create(&pp, &service_account).await {
                                 Ok(sa) => {
                                     println!(
                                         "Service Account : {} created in namespace : {}.",
                                         sa.metadata.name.unwrap(),
                                         sa.metadata.namespace.unwrap(),
-                                    );
+                                    )
                                 }
                                 Err(e) => {
                                     println!("Failed in creating service account  {:?}", e);
                                     tokio::time::sleep(Duration::from_secs(1)).await;
                                     std::process::exit(1);
                                 }
                             }
-                        } else {
+                        }
+                        Actions::Delete => {
+                            println!("service account does not exist!");
+                        }
+                    }
+                } else {
+                    match action {
+                        Actions::Create => {
                             println!(
                                 "Service Account : {} in namespace : {} already exist.",
-                                sa.items[0].metadata.name.as_ref().unwrap(),
-                                sa.items[0].metadata.namespace.as_ref().unwrap(),
+                                sa.items.get(0).unwrap().metadata.name.as_ref().unwrap(),
+                                sa.items
+                                    .get(0)
```

Comment:
you can tidy is by changing the if above to if let Some... and then you get the inner from the Some.